float32[] XX
float32[] YY
float32[] ZZ
